hardware/opentrons_hardware/hardware_control/tool_sensors.py

```diff
@@ -56,7 +57,9 @@ async def capacitive_probe(
     either by negating speed or negating distance.
     """
     sensor_scheduler = SensorScheduler()
-    sensor_info = SensorInformation(SensorType.capacitive, SensorId.S0, tool)
+    sensor_info = SensorInformation(
+        SensorType.capacitive, sensor_id or SensorId.S0, tool
+    )
```

Comment:
To match making `sensor_id` non-`Optional`: 

```suggestion
    sensor_info = SensorInformation(SensorType.capacitive, sensor_id, tool)
```

Reply:
Also, this is moot if we're going to adopt the above suggestion, but as a general rule: I think we should avoid expressions like `sensor_id or SensorId.S0`. Instead, we should do `sensor_id if sensor_id is not None else sensor_id`.

The `or` trick is more compact, but it makes it too easy to introduce bugs related to falsey values. For example, we've had bugs in the Python API where checks that were meant for `volume=None` accidentally trigger on `volume=0.0`. And, here, imagine if some future refactor changed the default from `SensorId.S0` to `SensorId.S1`. Then there would be a bug where passing `sensor_id=SensorId.S0` would erroneously default to `SensorId.S1`.

Reply:
True in general yes, but these are enums:
```
>>> class E(enum.Enum):
...   zero = 0
...   one = 1
... 
>>> E.zero
<E.zero: 0>
>>> bool(E.zero)
True
```

No element of `SensorId` is falsy, so this isn't a problem.

Reply:
`SensorId.S0` is falsey because `SensorId` subclasses `int`.

```python
>>> from opentrons_hardware.firmware_bindings.constants import SensorId
>>> bool(SensorId.S0)
False
```